Adds Eclipse config for xpack licence headers

colings86 opened the pull request:
This change adds to a step to the generation of eclipse files for each
x-pack project so that the code template for adding new files to x-pack
projects uses the Elastic Licence header rather than the Apache 2.0
licence header while non-xpack projects still use the Apache 2.0
licence header
x-pack/dev-tools/eclipse.settings/org.eclipse.jdt.ui.prefs

Note if testing this - In order to get eclipse to properly pick up this change and show the new template in the code templates UI (despite the `org.eclipse.jdt.ui.prefs` file being correct in the x-pack projects) I had to close Eclipse, run `git clean -xdf` and then `./gradlew eclipse` and then re-open Eclipse and refresh and clean all projects.

colings86 committed:
Adds Eclipse config for xpack licence headers

This change adds to a step to the generation of eclipse files for each
x-pack project so that the code template for adding new files to x-pack
projects uses the Elastic Licence header rather than the Apache 2.0
licence header while non-xpack projects still use the Apache 2.0
licence header
x-pack/dev-tools/eclipse.settings/org.eclipse.jdt.ui.prefs

elasticmachine commented:
Pinging @elastic/es-core-infra

rjernst commented on x-pack/build.gradle:
```diff
@@ -64,4 +65,13 @@ subprojects {
     }
   }
+  
+  task copyEclipseSettingsXPack(type: Copy) {
```
We already have a method to set settings, and it "works" here only because of the order of task execution (which could change). I would rather see this done in the existing eclipse settings task in the root build.gradle, and choose which (part) of the settings to copy. In particular, it would be good to not duplicate the other parts of the settings, which are already in `buildSrc/src/main/resources/eclipse.settings`. I think we should have the two licenses there, copy the files and append the appropriate code template line in a `doLast` on the copy?

colings86 committed:
iter

colings86 committed:
Moves code template logic into the root build.gradle

Uses a filter on the copy task for the eclipse settings files to
replace the token @@LICENSE_HEADER_TEXT@@ with the correct licence
header from the new buildSrc/src/main/resources/license-headers
directory

colings86 commented:
@rjernst I tried using a doLast block but couldn't get it working but I found a different way to get it working where we choose between the license headers based on the project name in the configuration phase and load the license header into a string. We then use a filter while copying the settings to replace the special toke `@@LICENSE_HEADER_TEXT@@` with the header text that we loaded before. This also has the advantage that all the code templates for eclipse remain the same across the projects except for this license text.

I'd be happy to change this to use a doLast block instead if you think thats still better but I think I'll need a bit of help getting it to work. The commit named "iter" in this PR shows what I tried with eh doLast block

rjernst commented on x-pack/build.gradle:
```diff
@@ -33,6 +33,7 @@ subprojects {
 }
 
 File checkstyleSuppressions = file('dev-tools/checkstyle_suppressions.xml')
+File eclipseSettings = file('dev-tools/eclipse.settings')
```
leftover?

rjernst reviewed (approved):
LGTM

colings86 committed:
Removes leftover line from previous implementation
